sockeye/inference.py

```diff
@@ -631,11 +644,51 @@ def _make_result(self,
                                 translation=target_string,
                                 tokens=target_tokens,
                                 attention_matrix=attention_matrix,
-                                score=neg_logprob)
+                                score=translation.score)
+
+    def _concat_translations(self, translations: List[Translation]) -> Translation:
+        """
+        Combine translations through concatenation.
+
+        :param translations: A list of translations (sequence, attention_matrix), score and length.
+        :return: A concatenation if the translations with a score.
+        """
+        # Concatenation of all target ids without BOS and EOS
+        target_ids = list(itertools.chain.from_iterable(translation.target_ids[1:-1] for translation in translations))
+        # Adding back the EOS symbol
+        target_ids = [self.vocab_target[C.BOS_SYMBOL]] + target_ids + [self.vocab_target[C.EOS_SYMBOL]]
```

Comment:
can we use itertools.chain here as well for speed?

Reply:
changed the code anyway to address michael's comment.